### PR TITLE
Added WestM keyboard 

### PR DESCRIPTION
**Pull request by WestMProducts**
Added support for a 65, and 68 key PCB, testing on prototypes showed features working correctly.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/westm/westm68/config.h [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x574d // WM
```
Comment (drashna) [inline keyboards/westm/westm68/config.h:20]:
```suggestion
#define VENDOR_ID       0x574D // WM
```

File: keyboards/westm/westm68/config.h [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+/* key matrix size */
+#define MATRIX_ROWS 5
+#define MATRIX_COLS 15
+
+#define MATRIX_ROW_PINS { A13, B9, F1, A10, A9 }
+#define MATRIX_COL_PINS { B14, B13, B12, B11, B10, B2, B1, B8, B7, B6, B5, B4, B3, A15, A14 }
+#define DIODE_DIRECTION COL2ROW
+
+#define RGBLIGHT_ANIMATIONS // Enable all additional animation modes.
```
Comment (drashna) [inline keyboards/westm/westm68/config.h:34]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/westm/westm68/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+	[0] = LAYOUT_65_ansi( /* Base */
```
Comment (drashna) [inline keyboards/westm/westm68/keymaps/via/keymap.c:20]: This needs to have 4 layers explicitly defined here.

Also, please replace the tabs with spaces.

File: keyboards/westm/westm68/rules.mk [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# MCU name
+MCU = STM32F072
+
+# Bootloader selection
+BOOTLOADER = stm32-dfu
+
+DFU_SUFFIX_ARGS = -v FFFF -p FFFF
+
+# Build Options
+BOOTMAGIC_ENABLE = lite     # Enable Bootmagic Lite
```
Comment (drashna) [inline keyboards/westm/westm68/rules.mk:10]: There is no "full" bootmagic anymore, only the "lite" option, which can be turned on with just "yes".
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

File: keyboards/westm/westm68/rules.mk [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
```
Comment (drashna) [inline keyboards/westm/westm68/rules.mk:15]:
```suggestion
```

File: keyboards/westm/westm68/rules.mk [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+NKRO_ENABLE = yes           # USB Nkey Rollover
+AUDIO_ENABLE = no           # Audio output
+
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no    # Breathing sleep LED during USB suspend
```
Comment (drashna) [inline keyboards/westm/westm68/rules.mk:20]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/westm/westm68/keymaps/via/keymap.c (hunk 3 above)
Comment (WestMProducts) [reply keyboards/westm/westm68/keymaps/via/keymap.c:20]: Hi Drashna, thanks for pointing on the other changes. I'm not exactly sure for this section what you mean by 4 layers? As in it should have 4 layers of other keymaps? Thanks!

Comment (drashna) [reply keyboards/westm/westm68/keymaps/via/keymap.c:20]: It needs 4 layers defined/filled out in the via keymap.  You only have 2.  If you don't fill out all of the layers needed, they can/will be filled with random/junk data in VIA

Comment (WestMProducts) [reply keyboards/westm/westm68/keymaps/via/keymap.c:20]: Okie! Added in the extra layers.

**Comment by drashna** [comment]
and could the same changes be made to the other revision/set of files, too

**Comment by WestMProducts** [comment]
> and could the same changes be made to the other revision/set of files, too

Oops, I missed a few changes on the Ergo board. Updated now.

File: keyboards/westm/westmergo/readme.md [hunk 7]
```diff
@@ -0,0 +1,14 @@
+# WestM
+
+![westmergo](https://imgur.com/tD395AG)
+
+A 65 key PCB with arrow cluster
+
+* Keyboard Maintainer: [WestM](https://github.com/westm00)
+* Hardware Supported: WestM65
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/westm/westmergo/readme.md:10]: Could you add instructions on how to enter the bootloader, to flash new firmware? 

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

File: keyboards/westm/westm68/readme.md [hunk 8]
```diff
@@ -0,0 +1,14 @@
+# WestM
+
+![westm68](https://imgur.com/EygC3BV)
+
+A 65% PCB with arrow cluster
+
+* Keyboard Maintainer: [WestM](https://github.com/westm00)
+* Hardware Supported: WestM65
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/westm/westm68/readme.md:10]: Could you add instructions on how to enter the bootloader, to flash new firmware? 

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/westm/westmergo/readme.md [hunk 9]
```diff
@@ -0,0 +1,14 @@
+# WestM
+
+![westmergo](https://imgur.com/tD395AG)
```
Comment (fauxpark) [inline keyboards/westm/westmergo/readme.md:3]:
```suggestion
![westmergo](https://i.imgur.com/tD395AG.png)
```

File: keyboards/westm/westmergo/readme.md [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+
+A 65 key PCB with arrow cluster
+
+* Keyboard Maintainer: [WestM](https://github.com/westm00)
+* Hardware Supported: WestM65
+
+Make example for this keyboard (after setting up your build environment):
+
+    make westm/westm65:default
```
Comment (fauxpark) [inline keyboards/westm/westmergo/readme.md:12]:
```suggestion
    make westm/westmergo:default
```

File: keyboards/westm/westm68/readme.md [hunk 11]
```diff
@@ -0,0 +1,14 @@
+# WestM
+
+![westm68](https://imgur.com/EygC3BV)
```
Comment (fauxpark) [inline keyboards/westm/westm68/readme.md:3]:
```suggestion
![westm68](https://i.imgur.com/EygC3BV.png)
```

File: keyboards/westm/westm68/readme.md [hunk 12]
```diff
@@ -0,0 +1,14 @@
...
+
+A 65% PCB with arrow cluster
+
+* Keyboard Maintainer: [WestM](https://github.com/westm00)
+* Hardware Supported: WestM65
+
+Make example for this keyboard (after setting up your build environment):
+
+    make westm/westm65:default
```
Comment (fauxpark) [inline keyboards/westm/westm68/readme.md:12]:
```suggestion
    make westm/westm68:default
```

File: keyboards/westm/westm68/rules.mk [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+NKRO_ENABLE = yes           # USB Nkey Rollover
+AUDIO_ENABLE = no           # Audio output
+
+
+BACKLIGHT_ENABLE = no	# Enable keyboard backlight(per switch) functionality
+RGB_MATRIX_ENABLE = no	# This is for per switch RGB which we are not using
+
+RGBLIGHT_ENABLE = yes	# Enable WS2812 RGB underglow.
```
Comment (fauxpark) [inline keyboards/westm/westm68/rules.mk:22]:
```suggestion
# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = yes        # Console for debug
COMMAND_ENABLE = yes        # Commands for debug and configuration
NKRO_ENABLE = yes           # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/westm/westmergo/rules.mk [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+NKRO_ENABLE = yes           # USB Nkey Rollover
+AUDIO_ENABLE = no           # Audio output
+
+BACKLIGHT_ENABLE = no	# Enable keyboard backlight(per switch) functionality
+RGB_MATRIX_ENABLE = no	# This is for per switch RGB which we are not using
+
+RGBLIGHT_ENABLE = yes	# Enable WS2812 RGB underglow.
```
Comment (fauxpark) [inline keyboards/westm/westmergo/rules.mk:21]:
```suggestion
# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = yes        # Console for debug
COMMAND_ENABLE = yes        # Commands for debug and configuration
NKRO_ENABLE = yes           # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/westm/westmergo/rules.mk [hunk 15]
```diff
@@ -0,0 +1,21 @@
+# MCU name
+MCU = STM32F072
+
+# Bootloader selection
+BOOTLOADER = stm32-dfu
+
+DFU_SUFFIX_ARGS = -v FFFF -p FFFF
```
Comment (fauxpark) [inline keyboards/westm/westmergo/rules.mk:7]:
```suggestion
# Wildcard to allow APM32 MCU
DFU_SUFFIX_ARGS = -v FFFF -p FFFF
```

File: keyboards/westm/westm68/rules.mk [hunk 16]
```diff
@@ -0,0 +1,22 @@
+# MCU name
+MCU = STM32F072
+
+# Bootloader selection
+BOOTLOADER = stm32-dfu
+
+DFU_SUFFIX_ARGS = -v FFFF -p FFFF
```
Comment (fauxpark) [inline keyboards/westm/westm68/rules.mk:7]:
```suggestion
# Wildcard to allow APM32 MCU
DFU_SUFFIX_ARGS = -v FFFF -p FFFF
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by WestMProducts** [comment]
@fauxpark @drashna I'm not sure if it's necessary but I also merged in the latest master changes, and updated the relevant changes required. Thank you!

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**